scripts/config_version.sh

```diff
@@ -1,14 +1,35 @@
 #!/bin/sh
 
+# Usage
 if [ $# -ne 2 -o ! -d "$1" -o ! -d "$1/$2" ]; then
   echo "usage: $0 <environmentpath> <environment>" >&2
   exit 1
 fi
 
-ruby=ruby
+# For portability, identify a preferred ruby executable to use
+ruby() {
+  [ -x /opt/puppetlabs/puppet/bin/ruby ] \
+    && /opt/puppetlabs/puppet/bin/ruby "$@" \
+    || /usr/bin/env ruby "$@"
+}
 
-if [ -x /opt/puppetlabs/puppet/bin/ruby ]; then
-  ruby=/opt/puppetlabs/puppet/bin/ruby
-fi
+# Determine how best to calculate a config_version
+if [ -e $1/$2/.r10k-deploy.json ]; then
+  # The environment was deployed using r10k. We will calculate the config
+  # version using the r10k data.
+  ruby $1/$2/scripts/config_version-r10k.rb $1 $2
+
+elif [ -e /opt/puppetlabs/server/pe_version ]; then
+  # This is a Puppet Enterprise system and we can rely on the rugged ruby gem
+  # being available.
+  ruby $1/$2/scripts/config_version-rugged.rb $1 $2
+
+elif [ -x /usr/bin/git ]; then
+  # The git command is available.
+  /usr/bin/git --git-dir $1/$2/.git rev-parse HEAD
```

Comment:
`git(1)` might be installed in another directory (e.g. `/usr/local/bin/git` on FreeBSD).  Maybe it would be better to assume the directory is in `$PATH` instead of using the full executable name?

Reply:
Good call. Updating to use `type git` to check for the command, and then expecting it on PATH.

Reply:
I was about to suggest another minor change and recall it's a branch in my repo. I ammended your commit to avoid relying on `test(1)` (`[(1)`) since `type` exit code tells us if the command was found :wink:

Reply:
Yep, you're totally right. Heh; oops.